File: 2005-check-if-all-the-integers-in-a-range-are-covered/check-if-all-the-integers-in-a-range-are-covered.cpp
class Solution {
public:
    bool isCovered(vector<vector<int>>& ranges, int left, int right) {
        for (int x = left; x <= right; x++) {
            bool covered = false;
            for (auto &r : ranges) {
                if (r[0] <= x && x <= r[1]) {
                    covered = true;
                    break;
                }
            }
            if (!covered) return false;
        }
        return true;
    }
};

File: 2005-check-if-all-the-integers-in-a-range-are-covered/check-if-all-the-integers-in-a-range-are-covered.py
class Solution(object):
    def isCovered(self, ranges, left, right):
        for x in range(left, right + 1):
            covered = False
            for start, end in ranges:
                if start <= x <= end:
                    covered = True
                    break
            if not covered:
                return False
        return True

        